.NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs

```diff
@@ -176,13 +177,45 @@ public bool GetMatchedTimeRange(string text, out string todSymbol, out int begin
                 beginHour = 20;
                 endHour = 23;
                 endMin = 59;
+
+            }
+            else if (((Options & DateTimeOptions.TasksMode) != 0) && DateTimeDefinitions.MealtimeBreakfastTermList.Any(o => trimmedText.Contains(o)))
```

Comment:
Are these four clauses changing some behaviour? Maybe I'm missing something from just reading the code, but they seem to just re-use the default resolution?

Reply:
These clauses are not changing any behaviour, I have added this clause since in comment at line no 229 in BaseDateTimeParser.cs,  it's mention that:  "Only standard time of day (morning, afternoon, evening and night) will not directly return".

Reply:
So, I concluded that following function GetMatchedTimeRange works only for standard time and not for mealtime which is specific to tasksmode.